Client/Frontend/Extensions/ClientPickerViewController.swift

```diff
@@ -172,91 +203,109 @@ class ClientPickerViewController: UITableViewController {
-                        self.refresh()
-                        return
-                    }
+        profile.remoteClientsAndTabs.getRemoteDevices() >>== { devices in
+            profile.remoteClientsAndTabs.getClients().uponQueue(.main) { result in
+                withExtendedLifetime(profile) {
+                    guard let clients = result.successValue else { return }
+                    self.update(devices: devices, clients: clients, endRefreshing: false)
+                    self.reloadClients()
                 }
-
-                self.reloadClients()
             }
-        })
+        }
     }
 
     fileprivate func reloadClients() {
-        let profile = self.ensureOpenProfile()
+        guard let profile = self.ensureOpenProfile() as? BrowserProfile, let account = profile.getAccount() else { return }
         self.loadState = .LoadingFromServer
 
-        profile.getClients().upon({ result in
-            withExtendedLifetime(profile) {
-                self.loadState = .Loaded
+        account.updateFxADevices(remoteDevices: profile.remoteClientsAndTabs) >>== {
+            profile.remoteClientsAndTabs.getRemoteDevices() >>== { devices in
+                profile.getClients().uponQueue(.main) { result in
+                    guard let clients = result.successValue else { return }
+                    withExtendedLifetime(profile) {
+                        self.loadState = .Loaded
 
-                // If we are running from an app extension then make sure we shut down the profile as soon as we are
-                // done with it.
-                if self.profileNeedsShutdown {
-                    profile._shutdown()
-                }
+                        // If we are running from an app extension then make sure we shut down the profile as soon as we are
+                        // done with it.
+                        if self.profileNeedsShutdown {
+                            profile._shutdown()
+                        }
 
-                self.loadState = .Loaded
-                guard let c = result.successValue else {
-                    return
-                }
+                        self.loadState = .Loaded
 
-                self.updateClients(clients: c, endRefreshing: true)
+                        self.update(devices: devices, clients: clients, endRefreshing: true)
+                    }
+                }
             }
-        })
+        }
     }
 
-    fileprivate func updateClients(clients: [RemoteClient], endRefreshing: Bool) {
+    fileprivate func update(devices: [RemoteDevice], clients: [RemoteClient], endRefreshing: Bool) {
+        assert(Thread.isMainThread)
+        guard let profile = self.ensureOpenProfile() as? BrowserProfile, let account = profile.getAccount() else { return }
+
+        let fxaDeviceIds = devices.compactMap { $0.id }
+        let devices = devices.filter { fxaDeviceIds.contains($0.id ?? "") }
```

Comment:
I don't think this line is needed. You're filtering against the same set of data so it should result in no change.

Reply:
Yeah this should be fixed. It is how filtering was done previously, but it doesn't look 100% right: https://github.com/mozilla-mobile/firefox-ios/pull/4682/files#diff-7a45ce0d70d55bdf297fcacd47e4e874L516

Reply:
Maybe it is correct, it filters out the null `id` devices.

Reply:
How about `let devices = devices.filter { $0.id != nil }` ?

Reply:
nvm, that isn't needed either

Reply:
`account.commandsClient.sendTab.isDeviceCompatible()` should be handling that case (a nil `id` returns false), but it isn't. I will add it there.